pkg/olm/clusterserviceversion.go

```diff
@@ -120,7 +120,44 @@ func (builder *ClusterServiceVersionBuilder) GetAlmExamples() (string, error) {
 		}
 	}
 
-	return "", fmt.Errorf("%s not found in given csv named %v", almExamples, builder.Definition.Name)
+	return "", fmt.Errorf("%s not found in given clusterServiceVersion named %v",
+		almExamples, builder.Definition.Name)
+}
+
+// IsSucceeded check if the clusterServiceVersion is Succeeded.
+func (builder *ClusterServiceVersionBuilder) IsSucceeded() (bool, error) {
+	if valid, err := builder.validate(); !valid {
+		return false, err
+	}
+
+	glog.V(100).Infof("Verify %s clusterServiceVersion in %s namespace is Succeeded",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	phase, err := builder.GetPhase()
+
+	if err != nil {
+		return false, fmt.Errorf("failed to get phase value for %s clusterserviceversion in %s namespace due to %w",
+			builder.Definition.Name, builder.Definition.Namespace, err)
+	}
+
+	return phase == "Succeeded", nil
+}
+
+// GetPhase get current clusterServiceVersion phase.
+func (builder *ClusterServiceVersionBuilder) GetPhase() (oplmV1alpha1.ClusterServiceVersionPhase, error) {
+	if valid, err := builder.validate(); !valid {
+		return "", err
+	}
+
+	glog.V(100).Infof("Get %s clusterServiceVersion in %s namespace phase",
```

Comment:
The above sentence doesn't look right.
Maybe: 
"Get %s clusterServiceVersion phase in namespace %s"